sample_apps/simple_app_under_islet/temporary_instructions.txt

```diff
@@ -116,61 +111,71 @@ make dummy
 make
 
 cd $CERTIFIER_PROTOTYPE/certifier_service
 
 go build simpleserver.go
 
-#run server
-cd $CCA_EXAMPLE_DIR/service
+# -----------------------------------------------------------------------------
+# NOTE: This arrangement is critical for this app to run cleanly.
+#       As we are working with a shim, simpleserver needs to be built with
+#       emulated key/cert bin files. So, provision them first.
+# provision service and apps
+# -----------------------------------------------------------------------------
+cd $ISLET_EXAMPLE_DIR/provisioning
+cp -p ./* $ISLET_EXAMPLE_DIR/service
+cp -p ./* $ISLET_EXAMPLE_DIR/app1_data
+cp -p ./* $ISLET_EXAMPLE_DIR/app2_data
+
+# Run server
+cd $ISLET_EXAMPLE_DIR/service
 
 $CERTIFIER_PROTOTYPE/certifier_service/simpleserver --policyFile=policy.bin --readPolicy=true &
 
 sleep 5
 
-# initialize client app
-cd $CCA_EXAMPLE_DIR
+# Initialize client app
+cd $ISLET_EXAMPLE_DIR
 
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true \
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true \
       --operation=cold-init --data_dir=./app1_data/ --measurement_file="example_app.measurement" \
       --policy_store_file=policy_store
 
 sleep 5
 
-# get client app certified
-# need to build libislet_sdk.so
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true \
+# Get client app certified (need to build libislet_sdk.so)
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true \
       --operation=get-certifier --data_dir=./app1_data/ --measurement_file="example_app.measurement" \
       --policy_store_file=policy_store
 
 sleep 5
 
-# initialize server app
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true \
+# Initialize server app
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true \
       --operation=cold-init --data_dir=./app2_data/ --measurement_file="example_app.measurement" \
       --policy_store_file=policy_store
 
 sleep 5
 
-# get server app certified
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true \
+# Get server app certified
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true \
       --operation=get-certifier --data_dir=./app2_data/ --measurement_file="example_app.measurement" \
-       --policy_store_file=policy_store
+      --policy_store_file=policy_store
 
 sleep 5
 
-#run the app
-cd $CCA_EXAMPLE_DIR
+# Run the app
+cd $ISLET_EXAMPLE_DIR
 
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true --operation=run-app-as-server --data_dir=./app2_data/ \
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true --operation=run-app-as-server --data_dir=./app2_data/ \
       --policy_store_file=policy_store --measurement_file="example_app.measurement" &
 
-sleep 5
+sleep 10
 
-cd $CCA_EXAMPLE_DIR
+cd $ISLET_EXAMPLE_DIR
 
-$CCA_EXAMPLE_DIR/cca_example_app.exe --print_all=true --operation=run-app-as-client --data_dir=./app1_data/ \
+$ISLET_EXAMPLE_DIR/islet_example_app.exe --print_all=true --operation=run-app-as-client --data_dir=./app1_data/ \
```

Comment:
This script now mostly works, but this very last step fails with a SSL-error:

```
+ /home/agurajada/Code/certifier-framework-for-confidential-computing/sample_apps/simple_app_under_islet/islet_example_app.exe --data_dir=./app1_data/ --operation=run-app-as-client --policy_store_file=policy_store --print_all=true
Running App as client
init_client_ssl: ssl_connect failed, ret=-1, err=1: SSL_ERROR_SSL: generic ssl error
Can't init client app
server_channel_accept_and_auth: Can't SSL_accept connection, res=-1
Accept error: SSL routines
SSL_ERROR_ZERO_RETURN: zero return error
```

It's unclear why this fails.

The very same steps implemented by `run_example.sh` work just fine.

I tried debugging this somewhat but could not find the problem. Will leave it for now as the driver bash script is working just fine.